For this integrated assignment, you will work in pairs to investigate a suspected Salmonella outbreak. The sequence data are real but the metadata (include the AMR information) are completely made up! We will use IRIDA to demonstrate how it can help with management of samples and preprocessing of sequence data.

Pairing Criteria:
1) Not in the same row
2) Not from the same organization
3) Not the same birth month
4) Ideally complementary backgrounds

IRIDA Demo:
Extensive documentation for users and developers
Start with a tutorial: https://irida.corefacility.ca/documentation/user/index.html#tutorials
1) Log-in (your user name is the first part of your email address (the part before the @ sign)
2) Basic navigation (Project, Samples, Analysis)
3) Upload a single dataset
3) Select samples and create new projects
4) Carry out some analysis and share that analysis 
Use this form to avoid redundant analysis, put your name next to the Strain you picked to analyze: https://docs.google.com/spreadsheets/d/1EX1z1JaV4hxm2n2ID5S2Az7CDGTs11_W6YFIzRmlqAk/edit#gid=1491924087
5) Access, view and download analysis results
6) Metadata

Scenerio:

Public health workers suspect a Canada-wide foodborne disease outbreak of Salmonella has occurred. Whole genome sequencing was performed on Salmonella isolates collected from  patients, food sources (retail), and environmental sources (farms and environmental sampling for surveillance). Some patient isolates showed resistance to a common antibiotic; other patient isolates showed resistance to multiple antibiotics.  The genomic data and epidemiological metadata have been uploaded to the IRIDA platform at Simon Fraser University (Project ID 10).  You are asked to perform phylogenomic analysis (tree pre-computed to save time), phylogeographic analysis, and temporospatial analysis on these sequences, and to interpret the results.  

What can you infer about the likelihood that one or more outbreaks have occurred? If you conclude that an outbreak has occurred, provide a hypothesis about how the outbreak occurred, including an identification of which isolates are outbreak associated (associated with human cases), and which ones are sporadic. Can you provide a hypothesis that explains the observation of antibiotic resistance in the isolates collected from human sources?

Tools to try: IRIDA (https://sfu.irida.ca), Microreact (https://microreact.org), Phandango (https://jameshadfield.github.io/phandango/#/), or iTOL (http://itol.embl.de/).

Don't forget to discuss with each other!
