---
layout: tutorial_page
permalink: /epidemiology_2018_Phylo&Vis
title: Genomic Epidemiology
header1: Emerging Pathogen Detection and Identification using Metagenomic Samples
header2: Infectious Disease Genomic Epidemiology
image: /site_images/CBW_wshop-epidem_map-icon.png
home: https://bioinformaticsdotca.github.io/epidemiology_2018
description: Emerging Pathogen Detection and Identification using Metagenomic Samples
author: Anamaria Crisan
modified: July 10th, 2018
---

# Tutorial 5 & 7 content

In this tutorial we will take a look at [GenGIS](http://kiwi.cs.dal.ca/GenGIS/Main_Page), [Microreact](https://microreact.org/showcase), [Phylocanvas](https://zachcp.github.io/phylocanvas/), and [R Shiny](https://shiny.rstudio.com/).

### Tutorial Data
We will primarily be using data from an Ebola dataset as well as a Klebsiella Pneumoniae dataset for the module 5 and module 7 tutorial components.

You can download both datasets to follow along was we trying out Microreact and Phylocanvas. For the shiny portion of the tutorial, the data has already been uploaded for you on the AWS instance

**Ebola Data**

[Ebola Paper](https://github.com/bioinformaticsdotca/GenEpi_2018/blob/master/module5_and_7/28405027.pdf)

[Ebola Metadata](https://raw.githubusercontent.com/bioinformaticsdotca/GenEpi_2018/master/module5_and_7/microreact-project-west-african-ebola-epidemic-data.csv)

[Ebola Tree](https://raw.githubusercontent.com/bioinformaticsdotca/GenEpi_2018/master/module5_and_7/microreact-project-west-african-ebola-epidemic-tree.nwk)

**Klebsiella Pneumoniaea**

[Kleb Pneumo Paper](https://github.com/bioinformaticsdotca/GenEpi_2018/blob/master/module5_and_7/1501049112.full.pdf)
    
[Kleb Pneumo Metadata](https://raw.githubusercontent.com/bioinformaticsdotca/GenEpi_2018/master/module5_and_7/kleb-pneumo-meta-data.csv)

[Kleb Pneumo Tree](https://raw.githubusercontent.com/bioinformaticsdotca/GenEpi_2018/master/module5_and_7/kleb-pneumo-meta-tree.nwk)


## GenGIS
GenGIS works on Windows and OSX machines, so to save time we will conduct a live demonstration and you will not need to download any files.


If you would like to follow along, you can download GenGIS and as well as [this map file](https://github.com/bioinformaticsdotca/GenEpi_2018/blob/master/module5_and_7/NaturalEarth2_WorldMap.tif).


## Phylocanvas and Microreact

We will upload the metadata and tree files to the PhyloCanvas and microreact websites and explore our data.

## R Shiny

We will go through a [brief introduction to R and Shiny](https://github.com/bioinformaticsdotca/GenEpi_2018/blob/master/module5_and_7/2018_Shiny_Intro.pdf).


Next, we'll take a look at a shiny application designed for Infectious Disease GenEpi, and that uses many of the visualizations deployed in GenGIS, Phylocanvas, and Microreact.

For those two are less comfortable with programming, try find someone in the class who is for this portion of the tutorial. Don't worry, there isn't too much for you to code. 

To logon to R studio go to:

 http://xx.oicrcbw.ca:8080 (xx is the student number)

 Username and password information will be distributed in class




    






